Tests/ManagerTests.swift

```diff
@@ -155,60 +156,108 @@ class ManagerTests: XCTestCase {
+        let resourceArtifacts = Manager.resourceArtifactsFromRequest(request)
 
         // Then
         XCTAssertEqual(resourceArtifacts!.ids!, [123], "Correct ID should be returned")
 
     }
 
-    func testResourceArtifactsFromURLShouldReturnMultipleIDs() {
+    func testResourceArtifactsFromRequestShouldReturnMultipleIDs() {
         // Given
-        let url = NSURL(string: "/path/123/to/resource/456")
+        let url = NSURL(string: "/path/123/to/resource/456")!
+        let request = NSURLRequest(URL: url)
 
         // When
-        let resourceArtifacts = Manager.resourceArtifactsFromURL(url!)
+        let resourceArtifacts = Manager.resourceArtifactsFromRequest(request)
 
         // Then
         XCTAssertEqual(resourceArtifacts!.ids!, [123, 456], "Two IDs, in order, should be returned")
     }
 
-    func testResourceArtifactsFromURLShouldReturnQueryParams() {
+    func testResourceArtifactsFromRequestShouldReturnQueryParams() {
         // Given
-        let url = NSURL(string: "/path/to/123/resource?foo=bar&baz=quux&foo=biz")
+        let url = NSURL(string: "/path/to/123/resource?foo=bar&baz=quux&foo=biz")!
+        let request = NSURLRequest(URL: url)
 
         // When
-        let resourceArtifacts = Manager.resourceArtifactsFromURL(url!)
+        let resourceArtifacts = Manager.resourceArtifactsFromRequest(request)
 
         // Then
         XCTAssertNotNil(resourceArtifacts!.query, "Query should be populated")
         XCTAssertEqual(resourceArtifacts!.query!, ["foo": ["bar", "biz"], "baz": ["quux"]], "Query dictionary should be correct")
     }
 
-    func testResourceArtifactsFromURLShouldNotReturnKeyForInvalidQueryParam() {
+    func testResourceArtifactsFromRequestShouldNotReturnKeyForInvalidQueryParam() {
         // Given
-        let url = NSURL(string: "/path/to/123/resource?foo=&bar=baz&123!!")
+        let url = NSURL(string: "/path/to/123/resource?foo=&bar=baz&123!!")!
+        let request = NSURLRequest(URL: url)
 
         // When
-        let resourceArtifacts = Manager.resourceArtifactsFromURL(url!)
+        let resourceArtifacts = Manager.resourceArtifactsFromRequest(request)
 
         // Then
         XCTAssertEqual(resourceArtifacts!.query!, ["bar": ["baz"]], "Returned dictionary should only have one value")
     }
 
-    func testResourceArtifactsFromURLShouldReturnNilForEmptyNSURL() {
+    func testResourceArtifactsFromRequestShouldReturnNilForEmptyNSURL() {
         // Given
-        let url = NSURL(string: "")
+        let url = NSURL(string: "")!
+        let request = NSURLRequest(URL: url)
 
         // When
-        let resourceArtifacts = Manager.resourceArtifactsFromURL(url!)
+        let resourceArtifacts = Manager.resourceArtifactsFromRequest(request)
 
         // Then
         XCTAssertNil(resourceArtifacts)
     }
+
+    // Skip this test for now due to a bug where HTTPBody is set to nil when PopTop receives the NSURLRequest
```

Comment:
As I understand it, the options are
- disable in Xcode scheme settings
- prefix with `x`
- comment the whole test

I like commenting the entire test because it's more obvious which test is being ignored.